api/v1alpha1/authenticationfilter_types.go

```diff
@@ -119,6 +128,58 @@ type RemoteJWKS struct {
 	// TODO: Add TBD remote JWKS fields based on defined use cases.
 }
 
+// OIDCAuthenticationFilterProvider defines the OpenID Connect (OIDC) authentication provider type
+type OIDCAuthenticationFilterProvider struct {
+	// The OIDC Provider configuration.
+	Provider OIDCProvider `json:"provider"`
+
+	// The OIDC client ID assigned to the filter to be used in the
+	// [Authentication Request](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest).
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	ClientID string `json:"clientId"`
```

Comment:
should this be part of a secret or is that not needed ?

Reply:
It's needed because it's required by the [OIDC Authorization Code Flow](https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest), and it's a standalone parameter, not part of the client secret.

- client_id REQUIRED.  The client identifier issued to the client during the registration process described by [Section 2.2](https://openid.net/specs/openid-connect-core-1_0.html#ClientAuthentication).
- client_secret REQUIRED.  The client secret.